src/ankh/helm/helm.go

```diff
@@ -232,20 +232,39 @@ func createReducedYAMLFile(filename, key string) error {
 }
 
 func Template(ctx *ankh.ExecutionContext, ankhFile ankh.AnkhFile) (string, error) {
+	var err error
 	finalOutput := ""
+	isChartFlagged := ctx.Chart != ""
+
+	doTemplate := func(ctx *ankh.ExecutionContext, chart ankh.Chart, ankhFile ankh.AnkhFile, output string) (string, error) {
+		ctx.Logger.Debugf("templating chart '%s'", chart.Name)
+		chartOutput, err := templateChart(ctx, chart, ankhFile)
+		if err != nil {
+			return output, err
+		}
+		output += chartOutput
+		return output, nil
+	}
 
 	if len(ankhFile.Charts) > 0 {
 		ctx.Logger.Debugf("templating charts")
+		m := make(map[string]ankh.Chart, len(ankhFile.Charts))
```

Comment:
i think the introduction of a map and the doTemplate inner function suggests it's more complex to try and generalize the single and multi chart case into one code path than it is to do the slightly dumber but more obvious thing.

i bet it'd be a smaller and more clearly correct if we separated these cases in an `if ctx.Chart != ""`block (which iterates the Charts, finds the matching one, and returns the result of templateChart immediately) and the `else` which maintains the existing code

Reply:
Yeah I tried it that way at first but it got messy when I had to keep track of if we found the name in the array or not with an extra var. Also, trying to continue through the loop when a name didn't match the chart was failing for me for some reason. 

I'll try refactoring to be more straightforward but with more duplicated code and misc variables...

Reply:
OK done, give it a look